plugin/pkg/admission/nodeenv/admission.go

```diff
@@ -0,0 +1,146 @@
+package nodeenv
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeEnvironment", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewPodNodeEnvironment(client, "")
+	})
+}
+
+const (
+	NamespaceNodeSelector = "kubernetes.io/node-selector"
+)
+
+// podNodeEnvironment is an implementation of admission.Interface.
+type podNodeEnvironment struct {
+	*admission.Handler
+	client              clientset.Interface
+	store               cache.Store
+	defaultNodeSelector string
```

Comment:
ok, if we want to get it from config, do we want to add a command line flag too to allow admins to configure the default?